e2e/framework.go

```diff
@@ -341,19 +385,17 @@ func (n *node) Start() {
 
 	go func() {
 	SYNC:
-		// 'sync up' with the network
-		_, history := n.c.syncWithNetwork(n.name)
-		n.proposals = history
-
+		_, syncIndex := n.c.syncWithNetwork(n.name)
+		n.setSyncIndex(syncIndex)
 		for {
 			fsm := &fsm{
 				n:            n,
 				nodes:        n.nodes,
-				lastProposer: n.lastProposer(),
+				lastProposer: n.c.getLastProposer(),
```

Comment:
I think this is not totally correct, what if due to partitions you do not sync up to the tip of the chain? You should get the proposal at your index and get the proposer from there. I would remove `getLastProposer` and do a `getProposalByIndex(indx uint64) *pbft.Proposal`. With this we can even remove `lastProposer` variable.

Reply:
Yes, I agree, this way we don't need additional `lastProposer` field in the cluster and also it will use `localIndex` to start sync from there.